packages/story-editor/src/components/colorPicker/header.js

```diff
@@ -29,46 +29,97 @@ import {
   BUTTON_VARIANTS,
 } from '@web-stories-wp/design-system';
 
-const HEADER_FOOTER_HEIGHT = 52;
+/**
+ * Internal dependencies
+ */
+import { focusStyle } from '../panels/shared';
+
+const HEADER_HEIGHT = 44;
 
 const Wrapper = styled.div`
   display: flex;
   justify-content: space-between;
   align-items: center;
-  height: ${HEADER_FOOTER_HEIGHT}px;
-  padding: 8px;
+  height: ${HEADER_HEIGHT}px;
+  padding: 4px 8px 4px 8px;
   position: relative;
+  border-bottom: 1px solid ${({ theme }) => theme.colors.divider.tertiary};
 `;
 
-const CloseButton = styled(Button)`
+const Actions = styled.div`
+  display: flex;
   margin-left: auto;
 `;
 
-function Header({ children, handleClose }) {
+const StyledButton = styled(Button)`
+  margin-left: 8px;
+  ${focusStyle};
+`;
+
+function Header({
+  children,
+  handleClose,
+  isEditMode = false,
+  setIsEditMode = () => {},
+  hasPresets = false,
+}) {
   const ref = useRef();
   useEffect(() => {
     ref.current?.focus();
   }, []);
+
+  const buttonProps = {
+    type: BUTTON_TYPES.TERTIARY,
+    size: BUTTON_SIZES.SMALL,
+    variant: BUTTON_VARIANTS.SQUARE,
+    onClick: (evt) => {
+      evt.stopPropagation();
+      setIsEditMode(!isEditMode);
+    },
+    isEditMode,
+  };
+
   return (
     <Wrapper>
       {children}
-      <CloseButton
-        aria-label={__('Close', 'web-stories')}
-        onClick={handleClose}
-        type={BUTTON_TYPES.TERTIARY}
-        size={BUTTON_SIZES.SMALL}
-        variant={BUTTON_VARIANTS.SQUARE}
-        ref={ref}
-      >
-        <Icons.Cross />
-      </CloseButton>
+      <Actions>
+        {hasPresets && isEditMode && (
+          <StyledButton
+            {...buttonProps}
+            aria-label={__('Exit edit mode', 'web-stories')}
+          >
+            {__('Done', 'web-stories')}
+          </StyledButton>
```

Comment:
This button is a bit tight (visible in the hover state mostly) - perhaps the button type should change when the wording is Done? Or we could change it to an icon - we have the floppy-disk save icon already, which might work?

![Screen Shot 2021-11-05 at 11 58 57](https://user-images.githubusercontent.com/637548/140541017-2f3afe2b-72bb-47f5-9331-5d7502e714ac.png)

Reply:
This is something that was brought directly over from the panel since the design for that state was missing in Figma. I'll follow up with Amy about it but leave it as is (and as was) for now.